src/include/distributed/master_protocol.h

```diff
@@ -85,6 +85,7 @@ extern void CheckDistributedTable(Oid relationId);
 extern void CreateShardPlacements(int64 shardId, List *ddlEventList,
 								  List *workerNodeList, int workerStartIndex,
 								  int replicationFactor);
+extern float4 UpdateShardStatistics(Oid relationId, int64 shardId);
```

Comment:
The return value feels pretty arbitrary. I'd feel slightly more comfortable returning the size or a statistics struct, and leave arbitrary calculations like fill factor to the caller.